#!/usr/bin/env python
PACKAGE = "robust_navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("maximum_linear_velocity",  			double_t,	0,  "Max speed Geobot can go",       0.5, 0, 1.0)
gen.add("maximum_angular_velocity", 			double_t,	0,  "Max spin speed Geobot will do", 0.4, 0, 1.1 )
gen.add("goal_tol",								double_t,	0,  "Fraction of current segment where next segment is considered",	0.1, 0, 1)
gen.add("num_of_segments_ahead",				int_t,		0,  "Number of future segments considered for speed reduction",		5,   1,	10)
gen.add("steering_gain",						double_t,	0,  "Steering gain term", 			1, 0, 2)
gen.add("error_gain",							double_t,	0,  "K_k Path offset gain term", 			1, 0, 2)
gen.add("cte_gain",								double_t,	0,  "cte Cross track error term", 			1, 0, 2)
gen.add("heading_gain",							double_t,	0,  "K_e Geobot heading error gain term", 	1, 0, 2)
gen.add("cross_track_error_deadband",			double_t,	0,	"cross track error deadband", 			0.02, 0, 1)
gen.add("derivative_cte_gain",					double_t,	0,  "derivative cte Cross track error term", 			0.1, 0, 2)
gen.add("derivative_heading_gain",				double_t,	0,  " derivative K_e Geobot heading error gain term", 	0.01, 0, 2)
gen.add("min_delta_time",						double_t,	0,  "min delta time", 	0, 0.1, 1)
exit(gen.generate(PACKAGE, "robust_navigation", "Gains"))
